include/gz/sim.hh

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GZ_SIM_HH_
+#define GZ_SIM_HH_
+
+// Since gz-cmake derives the auto-generated header name from the cmake
+// project name, for ignition-gazebo*, the auto-generated header name
+// is <gz/gazebo.hh>.
+// This header file provides a redirection from <gz/sim.hh> to <gz/gazebo.hh>
```

Comment:
Do we plan to have `gz/gazebo.hh` in gz-sim that forwards to `gz/sim.hh`? Otherwise, if users start using the header with `ign-gazebo3` or `ign-gazebo6` and then want to migrate to gz-sim, they will encounter build errors.

Reply:
yes, they would encounter build errors. I suppose we could add a `gz/gazebo.hh` header that redirects to `gz/sim.hh` and generates a deprecation warning when used

Reply:
it doesn't seem likely to me but it is possible